src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -336,6 +352,39 @@ protected async Task<ImmutableArray<Diagnostic>> VerifySourceGeneratorAsync(Solu
                     MatchDiagnosticsTimeout);
             }
 
+            // If any solution transforms for incremental generator testing have been defined,
+            // then test incremental generators.
+            if (IncrementalGeneratorTransforms.Count != 0)
+            {
+                var secondRunSolution = initialProject.Solution;
+                foreach (var incrementalTransform in IncrementalGeneratorTransforms)
+                {
+                    secondRunSolution = incrementalTransform(secondRunSolution, initialProject.Id);
+                }
+
+                var secondRunProject = secondRunSolution.GetProject(initialProject.Id)!;
+
+                updatedDriver = updatedDriver
+                        .ReplaceAdditionalTexts(secondRunProject.AnalyzerOptions.AdditionalFiles)
+                        .WithUpdatedParseOptions(secondRunProject.ParseOptions)
+                        .WithUpdatedAnalyzerConfigOptions(secondRunProject.AnalyzerOptions.AnalyzerConfigOptionsProvider());
```

Comment:
📝 I don't like how one of these transforms allows for any solution transform, but the generator driver transform only handles a few specific cases. It would be better if `IncrementalGeneratorTransforms` was a list of specific transforms that can be applied to the first compilation and first generator driver together.

Reply:
Ah, it seems like some of the generator properties are not part of the `Compilation` at all, so they are injected into the driver separately from the call to `RunGenerators`. Weird.

Reply:
I'll see if I can do a `ProjectState` object here like we do with the `CodeFixTest` code instead of doing solution transforms. Alternatively, we could give the users the `GeneratorDriver` and the `Compilation` and allow them to transform them themselves.